### ISSUE TITLE
X25519 is hard-coded as the default client key share algorithm

### ISSUE THREAD

**Issue by briansmith** [ctz/rustls#477]
I think the intent is that the choice of cipher suites is controlled by the user of Rustls. Then the user should be able to control which key exchange algorithms are enabled. Then they might not enable X25519, in which case even attempting to use X25519 would be wrong.

This is how the code looks in `choose_kx_groups`:
```
    let group = retryreq
        .and_then(HelloRetryRequest::get_requested_key_share_group)
        .or_else(|| find_kx_hint(sess, handshake.dns_name.as_ref()))
        .unwrap_or_else(|| NamedGroup::X25519);
```

Perhaps instead we should choose the first key exchange algorithm in the list of enabled key exchange algorithms, and document that this ordering matters and is used for this purpose?

**Comment by briansmith** [comment]
We should have a test where Rustls is configured to not have X25519 enabled to verify that this works.

**Comment by ctz** [comment]
This was fixed in #507
